### PR TITLE
fix for using PlasmaVesselDistance with ProximalProjection

### PR DESCRIPTION
**Pull request by dpanici**
using `ProximalProjection` with `surface_fixed=True` should be possible, this fixes the bug preventing that

### CODE REVIEW

**Comment by dpanici** [comment]
check if things only contains eq

File: desc/objectives/_geometry.py [hunk 1]
```diff
@@ -451,7 +451,7 @@ class PlasmaVesselDistance(_Objective):
         is fixed or not. If True, the surface is fixed and its coordinates are
         precomputed, which saves on computation time during optimization.
         If False, the surface coordinates are computed at every iteration.
-        False by default.
+        False by default, so that self.things = [eq] only, not [eq,surface]
```
Comment (f0uriest) [inline desc/objectives/_geometry.py:454]: I think this is the opposite?

File: desc/optimize/optimizer.py [hunk 2]
```diff
@@ -168,6 +168,15 @@ def optimize(  # noqa: C901 - FIXME: simplify this
             objective.build(verbose=verbose)
         if nonlinear_constraint is not None and not nonlinear_constraint.built:
             nonlinear_constraint.build(verbose=verbose)
+        # check and ensure ProximalProjection is only used with eq as an optimizable
+        if isinstance(objective, ProximalProjection):
+            if len(objective.things) > 1:
```
Comment (f0uriest) [inline desc/optimize/optimizer.py:173]: should also check that the one thing is an equilibrium

File: desc/objectives/_geometry.py (hunk 1 above)
Comment (ddudt) [reply desc/objectives/_geometry.py:454]: Oh yes, this needs to be changed. It should say something like: 

> If True, the surface is fixed and its coordinates are precomputed, which saves on computation time during optimization. In this case `self.things = [eq]` only. If False (default), the surface coordinates are computed at every iteration and `self.things = [eq, surface]`.

**Review by ddudt — CHANGES_REQUESTED** [review]

**Review by f0uriest — APPROVED** [review]

**Comment by f0uriest** [comment]
I think we should be able to get this to work even with `fixed_surface=False` (or with any other `thing`), the proximal part is more limited by what constraints than what objectives/things. I'll try to take a look at it.

**Review by ddudt — APPROVED** [review]

**Comment by f0uriest** [comment]
Going to merge this for now. I have some stuff that seems to work with `fixed_surface=False` but will need more testing and also required rolling in some changes from other PRs, so probably better to wait on that for a bit.

**Pull request merged**